langchain4j-qdrant/src/test/java/dev/langchain4j/store/embedding/qdrant/QdrantEmbeddingStoreIT.java

```diff
@@ -1,78 +1,1308 @@
+    private static int dimension = 384;
+    private static Distance distance = Distance.Cosine;
+    private static QdrantEmbeddingStore embeddingStore;
 
-  private static String collectionName = "langchain4j-" + randomUUID();
-  private static int dimension = 384;
-  private static int grpcPort = 6334;
-  private static Distance distance = Distance.Cosine;
-  private static QdrantEmbeddingStore embeddingStore;
+    @Container
+    private static final QdrantContainer qdrant = new QdrantContainer("qdrant/qdrant:v1.11.1");
 
-  @Container
-  private static final QdrantContainer qdrant = new QdrantContainer("qdrant/qdrant:latest");
+    EmbeddingModel embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
 
-  EmbeddingModel embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
-
-  @BeforeAll
-  static void setup() throws InterruptedException, ExecutionException {
-    embeddingStore =
-        QdrantEmbeddingStore.builder()
+    @BeforeAll
+    static void setup() throws InterruptedException, ExecutionException {
+        embeddingStore = QdrantEmbeddingStore.builder()
             .host(qdrant.getHost())
-            .port(qdrant.getMappedPort(grpcPort))
+            .port(qdrant.getGrpcPort())
             .collectionName(collectionName)
             .build();
 
-    QdrantClient client =
-        new QdrantClient(
-            QdrantGrpcClient.newBuilder(qdrant.getHost(), qdrant.getMappedPort(grpcPort), false)
-                .build());
-
-    client
-        .createCollectionAsync(
-            collectionName,
-            VectorParams.newBuilder().setDistance(distance).setSize(dimension).build())
-        .get();
-
-    client.close();
-  }
-
-  @AfterAll
-  static void teardown() {
-    embeddingStore.close();
-  }
-
-  @Override
-  protected EmbeddingStore<TextSegment> embeddingStore() {
-    return embeddingStore;
-  }
-
-  @Override
-  protected EmbeddingModel embeddingModel() {
-    return embeddingModel;
-  }
-
-  @Override
-  protected void clearStore() {
-    embeddingStore.clearStore();
-  }
-}
+        QdrantClient client = new QdrantClient(
+            QdrantGrpcClient.newBuilder(qdrant.getHost(), qdrant.getGrpcPort(), false)
+            .build());
+
+        client
+            .createCollectionAsync(
+                collectionName,
+                VectorParams.newBuilder().setDistance(distance).setSize(dimension)
+                .build())
+            .get();
+
+        client.close();
+    }
+
+    @AfterAll
+    static void teardown() {
+        embeddingStore.close();
+    }
+
+    @Override
+    protected EmbeddingStore < TextSegment > embeddingStore() {
+        return embeddingStore;
+    }
+
+    @Override
+    protected EmbeddingModel embeddingModel() {
+        return embeddingModel;
+    }
+
+    @Override
+    protected void clearStore() {
+        embeddingStore.clearStore();
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void should_filter_by_metadata_not(Filter metadataFilter,
```

Comment:
please extend this test class from `EmbeddingStoreWithFilteringIT` instead of duplicating test cases

Reply:
I've explained in the note of the PR description.

Reply:
Sorry, I did not notice it. Could you do something similar to what we did in `ChromaEmbeddingStoreIT.should_filter_by_metadata` and `ChromaEmbeddingStoreIT.should_filter_by_metadata_not`?

Reply:
I tried that too.
But there a quite a lot of cases to exclude.
So it made more sense to me to write the ones to include rather than exclude.

Reply:
@Anush008 it is hard to maintain duplicated test cases, please follow the same approach as in `ChromaEmbeddingStoreIT`. It should be simple given criterias are quite straightforward:

> Eq, NEq, In, NIn don't allow float and double values. Only integers and strings.
> LT, GT, LTE, GTE allow only numbers, not alphabets.
> For In and NIn conditions, if the key doesn't exist in the metadata, it is not matched.

Reply:
@langchain4j just FYI we've tested this PR in our environment and filtering now works fine for our cases

Reply:
@holeniev that's great! But please rewrite the test to reduce the duplication. Thank you!

Reply:
Thanks for checking @holeniev. I'll refactor the tests.

Reply:
Done.
[`f3fb841` (#1646)](https://github.com/langchain4j/langchain4j/pull/1646/commits/f3fb84142faef096bc0c60418e9c637f4b54764d)